config.go

```diff
@@ -80,3 +81,10 @@ func WithFilter(filter func(r *http.Request) bool) Option {
 		cfg.Filter = filter
 	})
 }
+
+// WithTraceResponseHeaderKey is used for changing response header key that contains trace id.
+func WithTraceResponseHeaderKey(name string) Option {
```

Comment:
Instead of using `name` as the parameter, I think it's better to use `func(name string) string` instead.

So if the user passes `nil`, we will use the default trace header, which is `X-Trace-ID`. However, if it's not, we will use the function to generate the header key name.